src/org/thoughtcrime/securesms/NewConversationActivity.java

```diff
@@ -75,19 +65,14 @@ private void handleIntent() {
         Uri uri = intent.getData();
         if(uri != null) {
           String scheme = uri.getScheme();
-          if(scheme != null && scheme.equals(MAILTO) ) {
-            String textToShare = getTextToShare(uri);
-            MailTo mailto = MailTo.parse(uri.toString());
-            String recipientsList = mailto.getTo();
-            if(recipientsList != null && !recipientsList.trim().isEmpty()) {
-              String[] recipientsArray = recipientsList.trim().split(",");
-              if (recipientsArray.length >= 1) {
-                String recipient = recipientsArray[0];
-                if (textToShare != null && !textToShare.isEmpty()) {
-                  getIntent().putExtra(TEXT_EXTRA, textToShare);
-                }
-                onContactSelected(DcContact.DC_CONTACT_ID_NEW_CONTACT, recipient);
+          if(MailtoUtil.isMailto(uri)) {
+            String textToShare = MailtoUtil.getText(uri);
+            String[] recipientsArray = MailtoUtil.getRecipients(uri);
+            if (recipientsArray.length >= 1) {
+              if (!textToShare.isEmpty()) {
+                getIntent().putExtra(TEXT_EXTRA, textToShare);
               }
+              onContactSelected(DcContact.DC_CONTACT_ID_NEW_CONTACT, recipientsArray[0]);
             } else {
```

Comment:
Nothing here changed semantically, you just extracted methods. Is my understanding correct?